### PR TITLE
template for data share

### PR DESCRIPTION
**Pull request by dadihe**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 101-data-share-share/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    "variables": {},
+    "resources": [
+        {
+            "type": "Microsoft.DataShare/accounts",
+            "apiVersion": "2019-11-01",
+            "name": "[parameters('account_name')]",
+            "location": "[parameters('location')]",
+            "identity": {
+                "type": "SystemAssigned"
```
Comment (bmoore-msft) [inline 101-data-share-share/azuredeploy.json:23]: how is this used?

File: 101-data-share-share/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "account_name": {
```
Comment (bmoore-msft) [inline 101-data-share-share/azuredeploy.json:5]: add metadata descriptions and appropriate defaultValues for all parameters

File: 101-data-share-share/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "account_name": {
```
Comment (bmoore-msft) [inline 101-data-share-share/azuredeploy.parameters.json:5]: you should be able to remove all the values from here once you have defaultValues in the template

File: 101-data-share-share/azuredeploy.json (hunk 1 above)
Comment (dadihe) [reply 101-data-share-share/azuredeploy.json:23]: > how is this used?

user has to enter their account and share name to create a share under that account.

File: 101-data-share-share/azuredeploy.parameters.json (hunk 3 above)
Comment (dadihe) [reply 101-data-share-share/azuredeploy.parameters.json:5]: > you should be able to remove all the values from here once you have defaultValues in the template

for the deployed locations we only support part of the regions of public and gov clouds.  Is there a way to keep them separate without having a parameter file?

File: 101-data-share-share/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-data-share-share/azuredeploy.json:26]: not the resource - but the MSI the resource is creating - how is that used?

File: 101-data-share-share/azuredeploy.parameters.json (hunk 3 above)
Comment (dadihe) [reply 101-data-share-share/azuredeploy.parameters.json:5]: @bmoore-msft

File: 101-data-share-share/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-data-share-share/azuredeploy.json:26]: How are permissions granted to the storage account?

Comment (dadihe) [reply 101-data-share-share/azuredeploy.json:26]: @bmoore-msft Data Share resource's managed identity needs to be granted access to Azure data store. Azure Data Share service then uses this managed identity to read and write data for snapshot-based sharing, and to establish symbolic link for in-place sharing. For more details you can refer to this link: https://docs.microsoft.com/en-us/azure/data-share/concepts-roles-permissions#roles-and-requirements

Comment (bmoore-msft) [reply 101-data-share-share/azuredeploy.json:48]: Got it - let's add this information to the readme so other users will know how it works as well...

File: 101-data-share-share/azuredeploy.json (hunk 2 above)
Comment (dadihe) [reply 101-data-share-share/azuredeploy.json:5]: descriptions have been added for all params

File: 101-data-share-share/azuredeploy.json (hunk 1 above)
Comment (dadihe) [reply 101-data-share-share/azuredeploy.json:48]: added.

File: 101-data-share-share/azuredeploy.parameters.json (hunk 3 above)
Comment (dadihe) [reply 101-data-share-share/azuredeploy.parameters.json:5]: Westus is not one of our support regions as of now, so I've added a 'allowedValues' for location with all supporting regions from public and usgov values. Please have a review for the new iteration. Thanks~

**Comment by dadihe** [comment]
@bmoore-msft for review

**Pull request merged**